src/lib/phdi-building-blocks/phdi_building_blocks/geo.py

```diff
@@ -1,119 +1,193 @@
+) -> Tuple[GeocodeResult, bool]:
+    """
+    Helper method to perform geocoding on a single address from a patient
+    resource. Here, the address is expressed in dictionary form, as it comes
+    straight out of a FHIR bundle. This function also determines whether the
+    geocoded standardized address is different from the raw address.
+    :param us_street.Client client: The API client to geocode with
+    :param dict address: A patient's address in FHIR / JSON
+    :return: Tuple containing the standardized geocoding result as well as
+    whether the information in the result is different than the information
+    held in the raw address
+    :rtype: Tuple[GeocodeResult, bool]
+    """
+
+    raw_one_line = get_one_line_address(address)
+    geocoded_result = get_geocoder_result(client, raw_one_line)
+
+    # Determine if standardized result is different from raw data
+    # Default assumption is addresses are same, in case there's no
+    # valid geocoder result
+    address_is_different = False
+    if geocoded_result:
+        std_one_line = f"{geocoded_result.address} {geocoded_result.city}, {geocoded_result.state} {geocoded_result.zipcode}"  # noqa
+        address_is_different = raw_one_line != std_one_line
 
-        raw_addresses = []
-        std_addresses = []
-        for address in patient.get("address", []):
-            # Generate a one-line address to pass to the geocoder
-            one_line = " ".join(address.get("line", []))
-            one_line += f" {address.get('city')}, {address.get('state')}"
-            if "postalCode" in address and address["postalCode"]:
-                one_line += f" {address['postalCode']}"
-            raw_addresses.append(one_line)
-
-            geocoded = geocode(client, one_line)
-            std_one_line = ""
-            if geocoded:
-                address["line"] = geocoded.address
-                address["city"] = geocoded.city
-                address["state"] = geocoded.state
-                address["postalCode"] = geocoded.zipcode
-                std_one_line = f"{geocoded.address} {geocoded.city}, {geocoded.state} {geocoded.zipcode}"  # noqa
-                std_addresses.append(std_one_line)
-
-                if "extension" not in address:
-                    address["extension"] = []
-
-                address["extension"].append(
-                    {
-                        "url": "http://hl7.org/fhir/StructureDefinition/geolocation",
-                        "extension": [
-                            {"url": "latitude", "valueDecimal": geocoded.lat},
-                            {"url": "longitude", "valueDecimal": geocoded.lng},
-                        ],
-                    }
-                )
-        any_dffs = (len(raw_addresses) != len(std_addresses)) or any(
-            [raw_addresses[i] != std_addresses[i] for i in range(len(raw_addresses))]
+    return (geocoded_result, address_is_different)
+
+
+def standardize_addresses_for_patient(
```

Comment:
I'm having a hard time following all of the various functions that are getting called for our geocoding. We're calling functions `standardize.*` when they're also geocoding, and geocoders are returning more than the geocoded results. I think this all needs to be reworked to make it more readable and intuitive.

Reply:
After speaking with Brandon, I believe we've cleaned this up a bit and agreed that it's fine.